test/integration/integration.cc

```diff
@@ -431,11 +431,20 @@ void BaseIntegrationTest::createGeneratedApiTestServer(const std::string& bootst
     // needs to know about the bound listener ports.
     auto end_time = time_system_.monotonicTime() + TestUtility::DefaultTimeout;
     const char* success = "listener_manager.listener_create_success";
+    // Loading invalid configuration may result in different error counters to be incremented.
+    // Loading invalid configuration using the google.protobuf.Any results in the
+    // "listener_manager.lds.update_failure" incremented. Loading invalid configuration using
+    // google.protobuf.Struct (deprecated in API v2) results in the
+    // "listener_manager.lds.update_rejected" incremented. See FilesystemSubscriptionImpl::refresh()
```

Comment:
This seems kind of wrong though, is it reasonable to easily change the behavior of typed config to be consistent with Struct? AFAICT it should be a rejection rather than a failure ideally.

Reply:
I've changed the exception type that is thrown when unknown fields are detected. So the update_rejected counter is incremented for the typed configs as well.